### DISCUSSION TITLE
Feature request: execute statement without sql editor selection

### DISCUSSION THREAD

**Discussion by ced75** [dbeaver/dbeaver#562] [Q&A] [answered]
Hi,

One thing that is mandatory in every SQL script is to select the whole SQL code before execute it.
Sometimes, when the SQL command is written on a couple of lines, the selection is painful. For example, you're in a transaction and you forget the first line in your selection => you get an error and have to rollback.

I don't know if it is hard to code, but what I would find very cool is the ability for DBeaver to execute the statement where the focus is in, without having to select it all. One statement could be defined as text between two semicolons.

Example:
![image](https://cloud.githubusercontent.com/assets/12892581/16448149/417b457e-3df0-11e6-9750-97febaa0b0c2.png)
Here, I have two statements: the red one and the blue one.
If the cursor is somewhere inside the red statement, the execution automatically selects the red code and executes it. If it is in the blue statement, it is this one that is executed.

What do you think about such a feature?

**Comment by serge-rider** [comment, marked as answer]
Usually you don't need to select query to execute it. Just point cursor on a query and press ctrl+enter.
DBeaver extracts queries separated by delimiter (";" symbol by default) and/or blank lines.

**Comment by ced75** [comment]
Great! I didn't notice it.
So, it's already done.
Thanks a lot.
